magma/wire_container.py

```diff
@@ -201,7 +210,10 @@ def wired(self):
 
     # return the input or output Bit connected to this Bit
     def trace(self, skip_self=True):
-        return self._wire.trace(skip_self)
+        try:
```

Comment:
I would say this is an ok way to do it, but perhaps needs better logic to actually detect a combo loop. not sure the best way to that...

Reply:
Actually a more precise way might be to keep a list of the current trace and error if we encounter a seen value, will try that.

Reply:
To be fair, I don't think this is a bad way to do this. Letting the system take care of tracking those values for us...

Reply:
It technically requires more stack frames than necessary (e.g. we could catch the loop in three recursive calls versus waiting for the RecursionError), but now that I think about it, given this is the very uncommon case, it doens't seem worth it to add extra cost here just for that error.